### ISSUE TITLE
Best way to integrate with fastai?

### ISSUE THREAD

**Issue by jph00** [qubvel/segmentation_models.pytorch#603]
@qubvel -- we've been chatting on the fast.ai Discord channel about how cool this lib is, and how much our community loves it! I'm wondering whether there might be ways to integrate it into fastai.

I'm not sure quite the best approach (or whether it's doable at all) but I'd be interested in exploring it. There's quite a bit of overlapping functionality between our libraries at the moment. We also integrate with timm, like you do, but currently we only do so for classification models, not for segmentation. fastai was (AFAIK) the first library to have a dynamic unet that supports a variety of pretrained backbones -- but you've taken it much further! I like the way you've got separate encoder and decoder modules, and also support various metrics and loss functions.

Anywho, I don't have any particularly concrete ideas - just wanted to raise the idea and see if you have any thoughts.

**Comment by jph00** [comment]
@qubvel -- just checking whether you had any thoughts on this?

**Comment by qubvel** [comment]
Hi @jph00 

Thanks for connecting! I am very sorry for not answering before, the project has not been maintained for a while and I had some other vital issues. 

I heard a lot about fastai, and it is widely used on Kaggle, but unfortunately never used it by myself. I guess we can explore the possibility of integration, it can be a nice feature for both libraries! If you have some thoughts about how to do it I am open to this! On my side, I can provide additional API for model creation if you have some standard one. Looking forward to hearing back from you, thanks!
